packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx

```diff
@@ -136,28 +153,46 @@ export const MafsGraph = (props: MafsGraphProps) => {
                         >
                             {/* Svg definitions to render only once */}
                             <SvgDefs />
-                            {/* Background layer */}
-                            <Grid
-                                tickStep={props.step}
-                                gridStep={props.gridStep}
-                                range={state.range}
-                                containerSizeClass={props.containerSizeClass}
-                                markings={props.markings}
-                            />
-                            {/* Locked figures layer */}
-                            {props.lockedFigures && (
-                                <GraphLockedLayer
-                                    lockedFigures={props.lockedFigures}
+                            {/* Cartesian grid nested in an SVG to lock to graph bounds */}
+                            <svg {...nestedSVGAttributes}>
+                                <Grid
+                                    gridStep={props.gridStep}
                                     range={state.range}
+                                    containerSizeClass={
+                                        props.containerSizeClass
+                                    }
+                                    markings={props.markings}
+                                    width={width}
+                                    height={height}
                                 />
-                            )}
-                            {/* Protractor */}
-                            {props.showProtractor && <Protractor />}
-                            {/* Interactive layer */}
-                            {renderGraph({
-                                state,
-                                dispatch,
-                            })}
+                            </svg>
+                            {/* Axis Ticks, Labels, and Arrows */}
+                            {
+                                // Only render the axis ticks and arrows if the markings are set to a full "graph"
+                                props.markings === "graph" && (
+                                    <>
+                                        <AxisTicks />
+                                        <AxisArrows />
+                                    </>
```

Comment:
Both the Axis Ticks and Axis Arrows need to be allowed to render outside of the bounds of the graph, but ALSO need to render below the protractor/interactive/locked figures AND above the gridlines. 

Unfortunately this requires the use of two separate nested SVGs so that we can render these elements in the correct order.